### PR TITLE
feat(ext/net): `Deno.resolveDns()` won't panic for not supported type

### LINKED ISSUE #14373: Should `Deno.resolveDns` handle error instead of panicked?
From #14355. For others invalid or typo in code, should Deno fall to `_ => {_}` case instead of panicking the application? The way to implements `_` case I thought is..

1. Print info as missing typo in code?
2. Print info as this is unsupported DNS record type and link to opening issue on this repo.

After that `_` case would return for `None` instead of `Some`.

### PR DESCRIPTION
**Pull request by PatrickChoDev**
Closes #14373

### CODE REVIEW

**Review by cjihrig — COMMENTED** [review]
Are you able to add a test?

File: ext/net/ops.rs [hunk 1]
```diff
@@ -704,7 +705,16 @@ where
     .filter_map(rdata_to_return_record(record_type))
     .collect();
 
-  Ok(results)
+  match results[0] {
+    DnsReturnRecord::Unsupported => {
+      let var: &'static str = record_type.into();
+      Err(custom_error(
+        "NotSupported",
+        var.to_owned() + " records are not supported for now",
```
[the remark this answers is not in the document]
Comment (PatrickChoDev) [reply ext/net/ops.rs]: Done

**Comment by PatrickChoDev** [comment]
> Are you able to add a test?
Since It's an unsopprted case. I think this'll be an integration test only?

**Comment by cjihrig** [comment]
> I think this'll be an integration test only?

I think that makes sense.

**Review by bartlomieju — CHANGES_REQUESTED** [review]
This looks good, but we really need a test to land it

File: cli/tests/testdata/resolve_dns.ts.out [hunk 2]
```diff
@@ -23,3 +23,4 @@ SRV
 TXT
 [["I","am","a","txt","record"],["I","am","another","txt","record"],["I am a different","txt record"],["key=val"]]
 Error NotFound thrown for not-found-example.com
+Error NotFound thrown for not-found-example.com
```
Comment (cjihrig) [inline cli/tests/testdata/resolve_dns.ts.out:26]: I would expect to see something different here. This test should be validating unsupported record type behavior.

File: ext/net/ops.rs [hunk 3]
```diff
@@ -704,7 +705,16 @@ where
     .filter_map(rdata_to_return_record(record_type))
     .collect();
 
-  Ok(results)
+  match results[0] {
+    DnsReturnRecord::Unsupported => {
+      let var: &'static str = record_type.into();
+      Err(custom_error(
+        "NotSupported",
+        var.to_owned() + " records are not supported",
```
Comment (cjihrig) [inline ext/net/ops.rs:713]: Are either of these showing up in the test added by this PR?

File: cli/tests/testdata/resolve_dns.ts.out (hunk 2 above)
Comment (PatrickChoDev) [reply cli/tests/testdata/resolve_dns.ts.out:26]: I'm not sure what the message should look like, could you provide me some?

File: ext/net/ops.rs (hunk 3 above)
Comment (PatrickChoDev) [reply ext/net/ops.rs:713]: This won't be appear in ts file but will display on Deno CLI, as I checked.

File: cli/tests/testdata/resolve_dns.ts.out (hunk 2 above)
Comment (cjihrig) [reply cli/tests/testdata/resolve_dns.ts.out:26]: The text here was created on lines 69-73 of the test added in this PR. Based on what this PR is supposed to be changing, I would expect the message to indicate that the record type is not supported. The message should ideally come from the runtime itself and not a string in the test file.

File: cli/Cargo.toml [hunk 4]
```diff
@@ -120,11 +120,12 @@ winapi = { version = "=0.3.9", features = ["knownfolders", "mswsock", "objbase",
 [dev-dependencies]
 deno_bench_util = { version = "0.51.0", path = "../bench_util" }
 flaky_test = "=0.1.0"
+openssl = { version = "0.10", features = ["vendored"] }
```
Comment (ry) [inline cli/Cargo.toml:123]: We are not going to add OpenSSL dependency.

Comment (lucacasonato) [reply cli/Cargo.toml:123]: More helpful feedback: use `rustls` to generate the RSA key. You can view some code in `ext/crypto` for how this is done there.

Comment (PatrickChoDev) [reply cli/Cargo.toml]: I've found that `trust-dns` also have builtin function to generate RSA keys but it's on openssl lib inside.

File: Cargo.lock [hunk 5]
```diff
@@ -1088,6 +1088,7 @@ dependencies = [
  "deno_core",
  "deno_tls",
  "log 0.4.17",
+ "openssl",
```
Comment (cjihrig) [inline Cargo.lock:1091]: I think all of these OpenSSL related changes need to be reverted before this can land.

**Review by cjihrig — CHANGES_REQUESTED** [review]

File: ext/net/Cargo.toml [hunk 6]
```diff
@@ -17,8 +17,9 @@ path = "lib.rs"
 deno_core = { version = "0.140.0", path = "../../core" }
 deno_tls = { version = "0.45.0", path = "../tls" }
 log = "0.4.16"
+openssl = { version = "0.10", features = ["vendored"] }
 serde = { version = "1.0.136", features = ["derive"] }
 socket2 = "0.4.4"
 tokio = { version = "1.17", features = ["full"] }
-trust-dns-proto = "=0.21.2"
+trust-dns-proto = { version = "=0.21.2" }
```
Comment (cjihrig) [inline ext/net/Cargo.toml:24]: It looks like this change is no longer necessary.

File: cli/tests/testdata/resolve_dns.ts [hunk 7]
```diff
@@ -62,3 +62,15 @@ try {
...
+  await Deno.resolveDns("example.com", "DNSKEY", nameServer);
+} catch (e) {
+  console.log(e.message);
+}
+
+try {
+  await Deno.resolveDns("example.com", "HINFO", nameServer);
+} catch (e) {
+  console.log(e.message);
+}
```
Comment (cjihrig) [inline cli/tests/testdata/resolve_dns.ts:76]: I think we only need one of these.

**Review by cjihrig — COMMENTED** [review]
It looks like this is still pulling in OpenSSL?

File: cli/tests/testdata/resolve_dns.ts (hunk 7 above)
Comment (PatrickChoDev) [reply cli/tests/testdata/resolve_dns.ts]: I'll remove DNSKEY then

File: ext/net/Cargo.toml (hunk 6 above)
Comment (PatrickChoDev) [reply ext/net/Cargo.toml]: I've forgetten to remove from main cargo.toml, sorry.

**Pull request closed without merging**